cmd/minishift/cmd/start_preflight.go

```diff
@@ -345,17 +345,44 @@ func checkLibvirtDefaultNetworkActive() bool {
 
 // checkHypervDriverSwitch returns true if Virtual Switch has been selected
 func checkHypervDriverSwitch() bool {
-	switchEnv := os.Getenv("HYPERV_VIRTUAL_SWITCH")
-	if switchEnv == "" {
+	switchEnvName := "HYPERV_VIRTUAL_SWITCH"
+	posh := powershell.New()
+	defer posh.Close()
+
+	switchEnvValue := os.Getenv(switchEnvName)
+
+	if switchEnvValue == "" {
+		checkIfDefaultSwitchExists := `Get-VMSwitch -Id c08cb7b8-9b3c-408e-8e30-5e16a3aeb444 | ForEach-Object { $_.SwitchType }`
+		fmt.Printf("\n   'Default Switch' ... ")
+		stdOut, _ := posh.Execute(checkIfDefaultSwitchExists)
+		if strings.Contains(stdOut, "Internal") {
+
+			// force setting the environment variable
+			os.Setenv("HYPERV_VIRTUAL_SWITCH", "Default Switch")
+			return true
+		}
+
 		return false
 	}
 
-	return true
+	if switchEnvValue != "" {
+		checkIfVirtualSwitchExists := fmt.Sprintf("Get-VMSwitch %s| ForEach-Object { $_.SwitchType }", switchEnvValue)
```

Comment:
This check is not case-sensitive, while the implementation in Docker machine seems to be:

```
PS> $env:HYPERV_VIRTUAL_SWITCH="dockernat"
-- Checking if Hyper-V driver is configured to use a Virtual Switch ...
   'dockernat' ... OK
-- Starting Minishift VM ..... FAIL E0117 13:28:15.889008   21556 start.go:366] Error starting the VM: Error creating the VM. Error with pre-create check: "vswitch \"dockernat\" not found". Retrying.
Error starting the VM: Error creating the VM. Error with pre-create check: "vswitch \"dockernat\" not found"
```

```
PS> $env:HYPERV_VIRTUAL_SWITCH="DockerNAT"
-- Checking if Hyper-V driver is configured to use a Virtual Switch ...
   'DockerNAT' ... OK
-- Starting Minishift VM ........................................... OK
```